tempodb/backend/block_meta.go

```diff
@@ -136,6 +160,118 @@ type BlockMeta struct {
 	ReplicationFactor uint32 `json:"replicationFactor,omitempty"`
 }
 
+func (b *BlockMeta) UnmarshalJSON(data []byte) error {
+	var msg interface{}
+	err := json.Unmarshal(data, &msg)
```

Comment:
instead of writing the below custom code can we do something like we did with dedicated cols:

https://github.com/grafana/tempo/blob/main/tempodb/backend/block_meta.go#L163

something like:

```
        type metaAlias BlockMeta

	err := json.Unmarshal(b, (*metaAlias)(b))
	if err != nil {
		return err
	}
	meta.Version = intern.Get(meta.Version)
        meta.TenantID = intern.Get(meta.TenantID)
```